tools/acorn-optimizer.js

```diff
@@ -1299,7 +1298,25 @@ if (extraInfoStart > 0) {
 // Collect all JS code comments to this array so that we can retain them in the outputted code
 // if --closureFriendly was requested.
 var sourceComments = [];
-var ast = acorn.parse(input, { ecmaVersion: 2018, preserveParens: closureFriendly, onComment: closureFriendly ? sourceComments : undefined });
+var ast;
+try {
+  ast = acorn.parse(input, {
+    ecmaVersion: 2018,
+    preserveParens: closureFriendly,
+    onComment: closureFriendly ? sourceComments : undefined
+  });
+} catch (err) {
+  err.message += (function() {
```

Comment:
Oh, I haven't seen this PR. Great job!

Nitpick: a function expression is unnecessary here, I'd just use block-scoped variables (or regular ones, if compatibility with old Node version is concern). But doesn't matter much :)

Reply:
Yeah I also like to use `let` or `const`.
But every variable declaration uses `var` in this repo.
So I follow it and use IIFE to void hoisting 😅